src/github.com/getlantern/lantern/feed.go

```
@@ -0,0 +1,170 @@
+// e.g. BBC, NYT, Reddit, etc.
+type Source struct {
+	FeedUrl string `json:"feedUrl"`
+	Title   string `json:"title"`
+	Url     string `json:"link"`
+	Entries []int  `json:"entries"`
+}
+
+type FeedItem struct {
+	Title       string                 `json:"title"`
+	Link        string                 `json:"link"`
+	Image       string                 `json:"image"`
+	Meta        map[string]interface{} `json:"meta,omitempty"`
+	Description string                 `json:"-"`
+}
+
+type FeedItems []FeedItem
+
+type FeedProvider interface {
+	AddSource(string)
+}
+
+type FeedRetriever interface {
+	AddFeed(string, string, string, string)
+}
+
+func FeedByName(name string, retriever FeedRetriever) {
+	if items, exists := feed.Items[name]; exists {
+		for _, i := range items {
+			retriever.AddFeed(i.Title, i.Description,
+				i.Image, i.Link)
+		}
+	}
+}
+
+// GetFeed creates an http.Client and fetches the latest
+// Lantern public feed for displaying on the home screen.
+// If a proxyAddr is specified, the http.Client will proxy
+// through it
+func GetFeed(locale string, proxyAddr string, provider FeedProvider) {
+	var err error
+	var req *http.Request
+	var res *http.Response
+
+	if !supportedLocales[locale] {
+		// always default to English if we don't
+		// have a feed available in a specific locale
+		locale = "en_US"
+	}
+
+	feedUrl := fmt.Sprintf(feedEndpoint, locale)
+
+	if req, err = http.NewRequest("GET", feedUrl, nil); err != nil {
+		log.Errorf("Error fetching feed: %v", err)
+		return
+	}
+
+	if proxyAddr == "" {
+		httpClient = &http.Client{}
+	} else {
+		httpClient, err = util.HTTPClient("", eventual.DefaultGetter(proxyAddr))
+		if err != nil {
+			log.Errorf("Error creating client: %v", err)
+			return
+		}
+	}
+
+	if res, err = httpClient.Do(req); err != nil {
+		log.Errorf("Error fetching feed: %v", err)
+		return
+	}
+
+	defer res.Body.Close()
+	contents, err := ioutil.ReadAll(res.Body)
+
+	if err != nil {
+		log.Errorf("Error reading body: %v", err)
```

If we're unable to load the feeds, shouldn't we let the client know instead of failing silently?